libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -16,29 +16,24 @@ impl Subpath {
 
 	/// Calculates the intersection points the subpath has with a given curve and returns a list of parameteric `t`-values.
 	/// This function expects the following:
-	/// - other: a [Bezier] curve to check intersections against
-	/// - error: an optional f64 value to provide an error bound
+	/// - `other`: a [Bezier] curve to check intersections against
+	/// - `error`: an optional f64 value to provide an error bound
+	/// - `minimum_seperation`: the minimum difference two adjacent `t`-values must have when comparing adjacent `t`-values in sorted order.
+	/// If the comparison condition is not satisfied, the function takes the larger `t`-value of the two.
 	/// <iframe frameBorder="0" width="100%" height="325px" src="https://graphite.rs/bezier-rs-demos#subpath/intersect-cubic/solo" title="Intersection Demo"></iframe>
-	pub fn intersections(&self, other: &Bezier, error: Option<f64>, minimum_seperation: Option<f64>) -> Vec<f64> {
+	pub fn intersections(&self, other: &Bezier, error: Option<f64>, minimum_seperation: Option<f64>) -> Vec<(usize, f64)> {
 		// TODO: account for either euclidean or parametric type
-		let number_of_curves = self.len_segments() as f64;
-		let intersection_t_values: Vec<f64> = self
+		let intersection_t_values: Vec<(usize, f64)> = self
 			.iter()
 			.enumerate()
-			.flat_map(|(index, bezier)| {
-				bezier
-					.intersections(other, error, minimum_seperation)
-					.into_iter()
-					.map(|t| ((index as f64) + t) / number_of_curves)
-					.collect::<Vec<f64>>()
-			})
+			.flat_map(|(index, bezier)| bezier.intersections(other, error, minimum_seperation).into_iter().map(|t| (index, t)).collect::<Vec<(usize, f64)>>())
 			.collect();
 
-		intersection_t_values.iter().fold(Vec::new(), |mut accumulator, t| {
-			if !accumulator.is_empty() && (accumulator.last().unwrap() - t).abs() < minimum_seperation.unwrap_or(MIN_SEPERATION_VALUE) {
+		intersection_t_values.iter().fold(Vec::<(usize, f64)>::new(), |mut accumulator, (index, t)| {
+			if !accumulator.is_empty() && (accumulator.last().unwrap().1 - t).abs() < minimum_seperation.unwrap_or(MIN_SEPERATION_VALUE) {
 				accumulator.pop();
 			}
-			accumulator.push(*t);
+			accumulator.push((*index, *t));
 			accumulator
 		});
```

Comment:
Thinking about this a bit more, I'm not sure that the fold here is necessary.
1. Doesn't fold return a value? We're not actually capturing the return value of the fold, so I'm not confident that this line is actually doing anything. I don't think fold actually modifies things in place
2. The fold behavior is basically removing points that are within `MIN_SEPERATION_VALUE` of each other. But, doesn't Bezier::intersection already do that? The only points that wouldn't be `MIN_SEPERATION_VALUE` sequential are those that might be on different segments, but we don't really care about that case, the whole point of adding `MIN_SEPERATION_VALUE` is to neatly handle the case where two segments overlap and generate too many intersections.

TLDR: I think we can just remove the whole `intersection_t_values.iter().fold(...);`